crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs

```diff
@@ -92,15 +94,23 @@ impl BasicBlock {
     /// Returns the terminator of this block, panics if there is None.
     ///
     /// Once this block has finished construction, this is expected to always be Some.
-    pub(crate) fn unwrap_terminator(&self) -> &TerminatorInstruction {
-        self.terminator().expect("Expected block to have terminator instruction")
+    pub(crate) fn unwrap_terminator(&self) -> Result<&TerminatorInstruction, InternalError> {
+        match self.terminator() {
+            Some(terminator_instruction) => Ok(terminator_instruction),
+            None => Err(InternalError::NoTerminatorInstructionInBlock { location: None }),
+        }
     }
 
     /// Returns a mutable reference to the terminator of this block.
     ///
     /// Once this block has finished construction, this is expected to always be Some.
-    pub(crate) fn unwrap_terminator_mut(&mut self) -> &mut TerminatorInstruction {
-        self.terminator.as_mut().expect("Expected block to have terminator instruction")
+    pub(crate) fn unwrap_terminator_mut(
```

Comment:
"unwrap" implies we're not checking to make sure that the terminator exists so we can remove this error.